graph/pull.go

```diff
@@ -256,6 +261,11 @@ func (s *TagStore) pullImage(r *registry.Session, out io.Writer, imgID, endpoint
 	// FIXME: Try to stream the images?
 	// FIXME: Launch the getRemoteImage() in goroutines
 
+	layers := append(history, imgID)
+	s.graph.Retain(layers, imgID)
+	// We need to delay release [imgID] Until image tagged.
+	defer s.graph.Release(history, []string{imgID})
+
```

Comment:
I think this whole added section would become unnecessary with the suggested sessionID change.